hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/timeline/HoodieTimelineArchiver.java

```diff
@@ -117,6 +122,10 @@ public boolean archiveIfRequired(HoodieEngineContext context, boolean acquireLoc
       } else {
         LOG.info("No Instants to archive");
       }
+      if (success && timerContext != null) {
+        long durationMs = metrics.getDurationInMs(timerContext.stop());
```

Comment:
Can we move the metrics handling to the write client or the service client, the cleaning and rollback alreay follow this pattern.

Reply:
Yeah, need to think through what the return type should look like.